dart-sample/build.gradle

```diff
@@ -1,49 +1,76 @@
-apply plugin: 'com.android.application'
+apply plugin: 'com.jakewharton.butterknife'
 
-android {
-    compileSdkVersion versions.compileSdk
+buildscript {
+    ext.versions = [
+            'minSdk': 14,
+            'compileSdk': 26,
 
-    compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_1_7
-        targetCompatibility = JavaVersion.VERSION_1_7
-    }
+            'supportLibrary': '25.3.0',
+            'androidPlugin': '3.0.0',
+            'androidTools': '26.0.0',
+            'butterknife' : '8.8.1',
+            'release': '3.0.0',
+    ]
+
+    ext.deps = [
+            android: [
+                    'runtime': 'com.google.android:android:4.1.1.4',
+                    'gradlePlugin': "com.android.tools.build:gradle:${versions.androidPlugin}",
+                    'appCompat': "com.android.support:appcompat-v7:${versions.supportLibrary}"
+            ],
+            parceler: [
+                    'runtime': 'org.parceler:parceler-api:1.0.4',
+                    'compiler': 'org.parceler:parceler:1.0.4'
+            ],
+            dart: [
+                'runtime': "com.f2prateek.dart:dart:${versions.release}",
+                'compiler': "com.f2prateek.dart:dart-processor:${versions.release}"
+            ],
+            henson:[
+                'runtime': "com.f2prateek.dart:henson:${versions.release}",
+                'compiler': "com.f2prateek.dart:henson-processor:${versions.release}"
+            ],
+            butterknife:[
+                'runtime': "com.jakewharton:butterknife:${versions.butterknife}",
+                'compiler': "com.jakewharton:butterknife-compiler:${versions.butterknife}",
+                'annotations': "com.jakewharton:butterknife-annotations:${versions.butterknife}",
+                'gradlePlugin': "com.jakewharton:butterknife-gradle-plugin:${versions.butterknife}",
+                'gradlePluginSnapshot': 'com.jakewharton:butterknife-gradle-plugin:9.0.0-SNAPSHOT'
```

Comment:
I think it's dangerous to use different versions of BK's plugin and lib, can we use the snapshot every where ?

Reply:
why do we need to use the snapshot?

Reply:
faced the issue ```com.android.build.gradle.api.BaseVariant.getOutputs()Ljava/util/List```
And the solution is to use snapshot here - https://github.com/JakeWharton/butterknife/pull/1026